test/jdk/java/nio/channels/spi/SelectorProvider/inheritedChannel/Launcher.java

```diff
@@ -100,11 +101,15 @@ public static SocketChannel launchWithInetSocketChannel(String className,
                                                         String... args)
             throws IOException
     {
-        try (ServerSocketChannel ssc = ServerSocketChannel.open()) {
-            ssc.socket().bind(new InetSocketAddress(InetAddress.getLocalHost(), 0));
-            InetSocketAddress isa = new InetSocketAddress(InetAddress.getLocalHost(),
-                                                      ssc.socket().getLocalPort());
-            SocketChannel sc1 = SocketChannel.open(isa);
+        ServerSocketChannel ch;
+        try {
+            ch = ServerSocketChannel.open(INET);
+        } catch (Exception e) {
+            ch = ServerSocketChannel.open();
+        }
+        try (ServerSocketChannel ssc = ch) {
+            ssc.socket().bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+            SocketChannel sc1 = SocketChannel.open(ssc.getLocalAddress());
```

Comment:
Could we add some logging/printlns here to indicate the protocol family of `ch` or its local address, to aid in debugging if we ever have problems with the test (eg on IPv6 only systems)?

Reply:
Added; now it produces output like:
```
Using INET (IPv4) channel
Socket bound to /127.0.0.1:44179
```